protocol/binary/string_post_go120.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+//go:build go1.20
+// +build go1.20
+
+package binary
+
+import "unsafe"
+
+// ReadString reads a Thrift encoded string.
+func (sr *StreamReader) ReadString() (string, error) {
+	bs, err := sr.ReadBinary()
+	return unsafe.String(unsafe.SliceData(bs), len(bs)), err
```

Comment:
As the name implies, any use of `unsafe` should be used very carefully. I'd strongly recommend adding a comment that indicates what assumptions are being made and why this is safe.

In this case, `bs` returned from `ReadBinary` has no mutable references (it creates a copy in that method), so it seems safe.

Reply:
An alternate implementation which avoids unsafe by using `strings.Builder` (which internally uses unsafe) would also work, and avoiding separate implementations by version
```go
var b strings.Builder
io.Copy(&b, sr.reader, length)
return b.String()
```

Reply:
Thanks for the suggestion. I've added a comment explaining why "unsafe" is safe to use here.

In your alternate implementation, I'm assuming you meant io.CopyN (not io.Copy). If yes, that ends up being even slower than the original implementation. 

When copying from a bytes.Buffer to a strings.Builder, I understand that io.Copy would avoid an allocation and a copy. However, io.CopyN is unable to make those optimisations and allocates a staging area for the copy operation.  That ends up outweighing the efficiency of the strings.Builder's String() function.

Reply:
benchmark results are here - https://github.com/saurabhagrawal-86/thriftrw-go/pull/1

Reply:
@prashantv, the benchmark results and evaluation of io.CopyN looks fine to me. We'll wait for your follow up!